### PR TITLE
Dump TSM files to line protocol

### PR DESCRIPTION
**Pull request by toddboom**
This extends the `influx_inspect` tool to provide a mechanism for dumping TSM files into line protocol. Probably still needs some usability love (being able to specify an output directory, for example), but is functional as-is.

Sample output:
```
$ go install ./... && $GOPATH/bin/influx_inspect dump
Opening /Users/todd/.influxdb/data/_internal/monitor/2/000000001-000000001.tsm in database _internal
Writing 75 series to /tmp/influxdb183681029/_internal.lp
Opening /Users/todd/.influxdb/data/stress/default/1/000000004-000000003.tsm in database stress
Writing 100000 series to /tmp/influxdb179802528/stress.lp
Opening /Users/todd/.influxdb/data/stress/default/3/000000001-000000001.tsm in database stress
Writing 2 series to /tmp/influxdb653771391/stress.lp
```

- [x] Rebased/mergable
- [x] Tests pass
- [ ] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: cmd/influx_inspect/main.go [hunk 1]
```diff
@@ -129,7 +129,24 @@ func main() {
 			fmt.Printf("%v", err)
 			os.Exit(1)
 		}
-		cmdVerify(path)
+		cmdDump(path)
+	case "dump":
```
Comment (jwilder) [inline cmd/influx_inspect/main.go]: Maybe call this `export`?

File: cmd/influx_inspect/dump.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		filename := temp + "/" + database + ".lp"
+
+		of, _ := os.Create(filename)
+		defer of.Close()
+		w := bufio.NewWriter(of)
+
+		fmt.Println("Writing", reader.KeyCount(), "series to", filename)
+
+		for i := 0; i < reader.KeyCount(); i++ {
+			key, _ := reader.KeyAt(i)
```
Comment (jwilder) [inline cmd/influx_inspect/dump.go]: You'll need to use the type to format the line protocol value correctly.

File: cmd/influx_inspect/dump.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+			values, _ := reader.ReadAll(key)
+			split := strings.Split(key, "#!~#")
+			measurement, field := split[0], split[1]
+
+			for _, value := range values {
+				pairs := field + "=" + fmt.Sprintf("%v", value.Value())
```
Comment (jwilder) [inline cmd/influx_inspect/dump.go]: Needs a type switch to format the values correctly.  Floats need a '.`, integers need a trailing `i`, strings need to be surrounded w/ quotes and bools can be `T` or `F` I think.

**Comment by toddboom** [comment]
@jwilder good catches. i'll get those fixed and added today.

**Comment by toddboom** [comment]
@jwilder ok, addressed the feedback. let me know if this looks good and i'll squash and merge.

**Comment by jwilder** [comment]
Needs a changelog.  But 👍

**Comment by kfitzpatrick** [comment]
Where are the associated docs for this?

**Pull request merged**